src/Rector/Deprecation/FileDefaultScheme.php

```diff
@@ -0,0 +1,39 @@
+<?php declare(strict_types=1);
+
+namespace DrupalRector\Rector\Deprecation;
+
+use DrupalRector\Rector\Deprecation\Base\FunctionToImmutableConfigBase;
+use Rector\Core\RectorDefinition\CodeSample;
+use Rector\Core\RectorDefinition\RectorDefinition;
+
+/**
+ * Replaces the deprecated file_default_scheme function calls.
+ *
+ * @see https://www.drupal.org/node/3049030 for change record.
+ */
+final class FileDefaultScheme extends FunctionToImmutableConfigBase
+{
+    protected $deprecatedFunctionName = 'file_default_scheme';
+    protected $configObject = 'system.file';
+    protected $configName = 'default_scheme';
```

Comment:
These variables make sense with the names. Can you also define these in the base class to help people understand which ones are expected? You can look at `FunctionToServiceBase` for an example.